eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CcConfiguration.java

```diff
@@ -10,305 +17,394 @@
 import org.eclipse.core.runtime.preferences.IScopeContext;
 import org.osgi.service.prefs.BackingStoreException;
 
+import com.google.common.collect.Sets;
+
+import cc.codechecker.plugin.CodeCheckerNature;
+import cc.codechecker.plugin.Logger;
 import cc.codechecker.plugin.config.Config.ConfigTypes;
+import cc.codechecker.plugin.config.Config.ConfigTypesCommon;
+import cc.codechecker.plugin.config.Config.ConfigTypesProject;
 import cc.codechecker.plugin.runtime.CodeCheckEnvironmentChecker;
 import cc.codechecker.plugin.runtime.CodecheckerServerThread;
-import cc.codechecker.plugin.CodeCheckerNature;
-import cc.codechecker.plugin.config.CodeCheckerContext;
-import cc.codechecker.plugin.views.console.ConsoleFactory;
-
-import java.io.File;
-import java.util.HashMap;
-import java.util.Map;
-
-import cc.codechecker.plugin.Logger;
-import org.eclipse.core.runtime.IStatus;
 
+/**
+ * Stores and manages configurations related to projects and the plug-in.
+ */
 public class CcConfiguration {
 
-    //Logger
-    IProject project=null;
-    IEclipsePreferences projectPreferences=null;
-    IEclipsePreferences globalPreferences = null;
-
-    private  Map<ConfigTypes,String> defaults;//default config values
-    private  Map<ConfigTypes,String> configKeys;//textual values of config variables
+    private static Map<ConfigTypes, String> globalConfig;
+    private static IEclipsePreferences globalPreferences;
+    
+    private static final String STR_EMPTY = "";
+    private static final String TRUE = "";
+    private static final String SEP = ":";
+    
+    private static final String CODECHECKER_WORKSPACE = ResourcesPlugin.getWorkspace().getRoot().getLocation().toString() 
+            + "/.codechecker";
+
+    private static final Map<ConfigTypes, String> DEFAULT_CONFIG = new HashMap<>();
+    static {
+        DEFAULT_CONFIG.put(ConfigTypesCommon.COMPILERS,"gcc:g++:clang");
+        DEFAULT_CONFIG.put(ConfigTypesCommon.ANAL_THREADS,"4");
+        DEFAULT_CONFIG.put(ConfigTypesCommon.CHECKER_PATH,STR_EMPTY);
+        DEFAULT_CONFIG.put(ConfigTypesCommon.PYTHON_PATH,"");
+        // TODO Somehow set a default checker List OR in User Interface
+        // set a check-box to use the default checkers.
+        DEFAULT_CONFIG.put(ConfigTypesCommon.CHECKER_LIST,STR_EMPTY);
+    }
 
-    public void init() {        
-        globalPreferences = ConfigurationScope.INSTANCE.getNode(CodeCheckerNature.NATURE_ID);
-        defaults=new HashMap<ConfigTypes,String>();
-        defaults.put(ConfigTypes.COMPILERS,"gcc:g++:clang");
-        defaults.put(ConfigTypes.ANAL_THREADS,"4");
-        defaults.put(ConfigTypes.IS_GLOBAL,"true");
+    //Project related fields
+    private IProject project;
+    private boolean useGlobal = true;
+    // This configuration has the same keys as global.
+    private Map<ConfigTypesCommon,String> commonProjectConfig;
+    // These has no global meaning.
+    private Map<ConfigTypesProject, String> projectOnlyConfig;
```

Comment:
Maybe a slightly changed comment would be better here like `this is a project local configuration not available globally`